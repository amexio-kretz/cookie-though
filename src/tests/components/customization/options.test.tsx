import { h } from 'preact';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import mockCookies from '../../__mocks__/cookieOptions';

import Options from '../../../components/customization/options';
import Option from '../../../components/customization/option';

const defaultProps = {
  options: mockCookies,
  onToggle: jest.fn(),
};

describe('Option', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Options {...defaultProps} />);
    expect(wrapper.find(Option)).toHaveLength(3);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls the correct function when an option is toggled', () => {
    const onToggle = jest.fn();
    const wrapper = mount(<Options {...defaultProps} onToggle={onToggle} />);
    const option = wrapper.find(Option).first();
    option.find('button').simulate('click');

    expect(onToggle).toBeCalledTimes(1);
  });
});
