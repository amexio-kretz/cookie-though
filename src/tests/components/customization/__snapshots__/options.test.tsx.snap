// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option should render properly 1`] = `
<div
  className="options"
>
  <div
    className="option optionSecondary"
  >
    <div
      className="optionInfo"
    >
      <p>
        Stricly necessary cookies
      </p>
    </div>
    <div
      className="optionDefault"
    >
      Always on
    </div>
  </div>
  <Option
    onToggle={[Function]}
    option={
      Object {
        "description": "We’ll remember the basics such as language.",
        "isEnabled": false,
        "name": "Functional cookies",
      }
    }
  />
  <Option
    key={1}
    onToggle={[Function]}
    option={
      Object {
        "description": "We’ll know where we should improve your experience.",
        "isEnabled": false,
        "name": "Analytics",
      }
    }
  />
  <Option
    key={2}
    onToggle={[Function]}
    option={
      Object {
        "description": "We’ll only show you content that interests you.",
        "isEnabled": false,
        "name": "Marketing",
      }
    }
  />
  <div
    className="declaration"
  >
    <a
      href="#"
    >
      Read the full cookie declaration
    </a>
  </div>
</div>
`;
